File: 프로그래머스/0/181894. 2의 영역/2의 영역.java
class Solution {
    public int[] solution(int[] arr) {
        int[] answer = {};
        
        int start = -1;
        int end = -1;
        int index = 0;
        for(index = 0; index < arr.length; index++){
            if(arr[index] == 2){
                if(start != -1){
                    end = index;
                }
                else{start = index;}
            }
            
        }
        System.out.println("start = " + start);
        System.out.println("end = " + end);
        
        if(start == -1){
            return new int[]{-1};
        }
        if(start != -1 && end == -1) return new int[]{2};
        
        int[] temp = new int[end - start + 1];
        System.out.println(temp.length);
        for(int i = 0; i < temp.length; i++){
            temp[i] = arr[start + i];
        }
        
        
        return temp;
    }
}

File: 프로그래머스/0/181894. 2의 영역/2의 영역.js
function solution(arr) {
    var answer = [];
    if(!arr.includes(2)) return [-1];
    
    // const indexes = arr
    // .map((item, index) => {
    //     if(item === 2) return index;
    //     else return -1;
    // })
    // .filter((item, index) => (item !== -1));
    
    const ind = arr.reduce((a, c, index) => {
        if(c === 2){
            a.push(index);
        }
        return a;
    },[]);
    
    return arr.slice(ind[0], ind.at(-1) + 1);
    
}